api/tokens/serializers.py

```diff
@@ -73,11 +101,54 @@ def update(self, instance, validated_data):
                 continue
             else:
                 setattr(instance, attr, value)
+        if scopes:
+            update_scopes(instance, scopes)
+
         instance.save()
         return instance
 
-def validate_requested_scopes(validated_data):
-    scopes_set = set(validated_data.get('scopes', '').split(' '))
-    for scope in scopes_set:
-        if scope not in public_scopes or not public_scopes[scope].is_public:
-            raise exceptions.ValidationError('User requested invalid scope')
+
+class ApiOAuth2PersonalTokenWritableSerializer(ApiOAuth2PersonalTokenSerializer):
+    def __init__(self, *args, **kwargs):
+        super(ApiOAuth2PersonalTokenWritableSerializer, self).__init__(*args, **kwargs)
+        request = kwargs['context']['request']
+
+        # Dynamically overriding scopes field for early versions to make scopes writable via an attribute
+        if not expect_scopes_as_relationships(request):
+            self.fields['scopes'] = ser.CharField(write_only=True, required=False)
```

Comment:
Serializer added to preserve behavior of adding scopes as an attribute for versions pre-2.11.   Scopes need to be a CharField for the request, but a SerializerMethodField for the response.    This writable serializer allows the  former on requests.

Reply:
Should this affect the `owner` field as well? In `ApiOAuth2PersonalTokenSerializer `, we set `self.fields['owner']` as a `ser.SerializerMethodField()` if `expect_scopes_as_relationships(request)` is false. It was a char field before.